functions/functionsv2/hellostorage/hello_storage.go

```diff
@@ -21,40 +21,32 @@ import (
 	"context"
 	"fmt"
 	"log"
-	"time"
 
 	"github.com/GoogleCloudPlatform/functions-framework-go/functions"
 	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/googleapis/google-cloudevents-go/cloud/storagedata"
+	"google.golang.org/protobuf/encoding/protojson"
 )
 
 func init() {
 	functions.CloudEvent("HelloStorage", helloStorage)
 }
 
-// StorageObjectData contains metadata of the Cloud Storage object.
-type StorageObjectData struct {
-	Bucket         string    `json:"bucket,omitempty"`
-	Name           string    `json:"name,omitempty"`
-	Metageneration int64     `json:"metageneration,string,omitempty"`
-	TimeCreated    time.Time `json:"timeCreated,omitempty"`
-	Updated        time.Time `json:"updated,omitempty"`
-}
-
 // helloStorage consumes a CloudEvent message and logs details about the changed object.
 func helloStorage(ctx context.Context, e event.Event) error {
 	log.Printf("Event ID: %s", e.ID())
 	log.Printf("Event Type: %s", e.Type())
 
-	var data StorageObjectData
-	if err := e.DataAs(&data); err != nil {
-		return fmt.Errorf("event.DataAs: %v", err)
+	data := &storagedata.StorageObjectData{}
+	if err := protojson.Unmarshal(e.Data(), data); err != nil {
+		return fmt.Errorf("protojson.Unmarshal: %v", err)
```

Comment:
quibble: I would use the %w format here, allows someone debugging the sample to unwrap the error.